server/app/services/migration/ProgramMigrationService.java

```diff
@@ -70,4 +76,44 @@ public ErrorAnd<ProgramMigrationWrapper, String> deserialize(String programJson)
           ImmutableSet.of(String.format("JSON is incorrectly formatted: %s", e.getMessage())));
     }
   }
+
+  /**
+   * Checks if there are existing questions that match the admin id of any of the incoming
+   * questions. If a match is found, generate a new admin name of the format "orginal admin name-n".
+   *
+   * <p>Return a map of old_question_name -> updated_question_data
+   */
+  public ImmutableMap<String, QuestionDefinition> maybeOverwriteQuestionName(
+      ImmutableList<QuestionDefinition> questions) {
+    return questions.stream()
+        .collect(
+            ImmutableMap.toImmutableMap(
+                QuestionDefinition::getName,
+                question -> {
+                  String newAdminName = maybeGenerateNewAdminName(question.getName());
+                  try {
+                    return new QuestionDefinitionBuilder(question).setName(newAdminName).build();
+                  } catch (UnsupportedQuestionTypeException error) {
+                    throw new RuntimeException(error);
+                  }
+                }));
+  }
+
+  /**
+   * Generate a new admin name for questions of the format "orginal admin name-n" where "n" is the
+   * next consecutive number such that we don't already have a question with that admin name saved.
+   * For example if the admin name is "sample question" and we already have "sample question-1" and
+   * "sample question-2" saved in the db, the generated name will be "sample question-3"
+   */
+  public String maybeGenerateNewAdminName(String adminName) {
+    String newAdminName = adminName;
+    ImmutableList<String> similarAdminNames = questionRepository.getSimilarAdminNames(adminName);
```

Comment:
@dkatzz I opted to handle this here since we need to strip the suffix off to properly rename the questions anyway. I added a test case to `ProgramMigrationServiceTest` and left the method in `QuestionRepository` alone. What do you think think of this fix?

Reply:
SGTM, thank you!! Just one comment around the test, but otherwise looks good!